engine/src/main/java/org/hibernate/search/engine/impl/ImmutableAnalyzerRegistry.java

```diff
@@ -67,13 +77,19 @@ public AnalyzerReference getAnalyzerReference(Class<?> analyzerClazz) {
 
 	@Override
 	public void close() {
-		close( referencesByClass.values() );
-	}
-
-	private void close(Collection<AnalyzerReference> references) {
-		for ( AnalyzerReference reference : references ) {
+		for ( AnalyzerReference reference : getAllReferences() ) {
 			reference.close();
 		}
 	}
 
+	private List<AnalyzerReference> getAllReferences() {
```

Comment:
@Sanne I wanted to use a Stream, but decided against it in order to be able to backport.
I guess I'll add a good old closeAll() method.

> BTW should we invoke each close() method in a try/catch "log & ignore" block?
I'll try to do that. Ideally we should throw an exception if anything goes wrong and still close every reference, but in order to do that, we should do it everywhere, and unfortunately we don't: https://hibernate.atlassian.net/browse/HSEARCH-2640